protocol/src/types/primitive.rs

```diff
@@ -232,13 +232,21 @@ pub struct Metadata {
     pub cycles_limit:    u64,
     pub cycles_price:    u64,
     pub interval:        u64,
-    pub verifier_list:   Vec<Validator>,
+    pub verifier_list:   Vec<ValidatorExtend>,
     pub propose_ratio:   u64,
     pub prevote_ratio:   u64,
     pub precommit_ratio: u64,
     pub brake_ratio:     u64,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct ValidatorExtend {
```

Comment:
Why use a new struct here other than add the `bls_pub_key` field into the existing `Validator` struct?